src/storage/index.js

```diff
@@ -13,22 +13,26 @@ const s3 = new aws.S3({
   credentials: config.s3.credentials,
 });
 
-const getFileUpload = (fileTag, maxFileSize) => {
-  const fileUpload = multer({
-    storage: multerS3({
-      s3,
-      bucket: config.s3.bucket,
-      acl: 'public-read',
-      key: (req, file, next) => {
-        next(null, `portal/profiles/${req.user.uuid}${path.extname(file.originalname)}`);
-      },
-    }),
+const uploadToS3 = async (mediaType, file, uuid) => {
+  const mediaTypeConfig = Media.getMediaTypeConfig(mediaType);
+  const params = {
+    ACL: 'public-read',
+    Body: file.buffer,
+    Bucket: config.s3.bucket,
+    Key: `portal/${mediaTypeConfig.uploadPath}/${uuid}${path.extname(file.originalname)}`,
+  };
+  return s3.upload(params).promise();
+};
+
+const bufferImageBlob = (mediaType) => {
+  const mediaTypeConfig = Media.getMediaTypeConfig(mediaType);
+  return multer({
+    storage: multer.memoryStorage(),
     limits: {
-      fileSize: maxFileSize * BYTES_PER_KILOBYTE,
+      fileSize: mediaTypeConfig.maxFileSize * BYTES_PER_KILOBYTE,
     },
-  }).single(fileTag);
-  return fileUpload;
+  }).single(mediaTypeConfig.fileTag);
```

Comment:
Nit: `fileTag` is actually something the controller handles, since that's part of the request and it's getting passed into this service. Maybe include it as a parameter along with `mediaType`? So we still hide the `multer.single` call but have the controller specify from where in the request the file's coming from.